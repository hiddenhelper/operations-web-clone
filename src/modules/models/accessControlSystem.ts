import { ILocation } from './address';
import { IDevice, DeviceStatus } from './device';

export enum AccessControlSystemType {
  PORTAL = 0,
  TURNSTILE = 1,
  ENCLOSED_TS = 2,
  HANDHELD = 3,
}

export enum ReaderType {
  Entrance = 0,
  Exit = 1,
  BiDirectionalByAntenna = 2,
  BiDirectionalBySpec = 3,
}

export interface IReader {
  serialNumber: string;
  lastMaintenanceDate?: string;
  inServiceDate: string;
  model: string;
  notes?: string;
  hostname?: string;
  directionType?: ReaderType;
  hostAddress?: string;
  telnetConnectionPort?: string;
  sshConnectionPort?: string;
  httpConnectionPort?: string;
  tcpConnectionPort?: string;
}

export enum LifeCycleType {
  NEW = 0,
  UPGRADE = 1,
  IN_FIELD_TRANSFER = 2,
  REFURBISHED = 3,
}

export enum AccessControlSystemVersion {
  V1 = 0,
  V2 = 1,
  V3 = 2,
  V3_FACIAL = 3,
}

export interface IAccessControlSystem extends IDevice {
  type: AccessControlSystemType;
  lifeCycle: LifeCycleType;
  lastRefurbishedDate: string;
  lastMaintenanceDate: string;
  inServiceDate: string;
  warrantyExpirationDate: string;
  orderDate?: string;
  vendor?: string;
  invoice?: string;
  reader1?: IReader;
  reader2?: IReader;
  version: AccessControlSystemVersion;
  project?: {
    id: string;
    name: string;
  };
  status: DeviceStatus;
  location?: ILocation;
  hasReverseInstallation?: boolean;
  reader1SerialNumber?: string;
  reader2SerialNumber?: string;
  reader1Hostname?: string;
  reader2Hostname?: string;
}

export interface IProjectAccessControlSystem {
  id?: string;
  accessControlSystemId?: string;
  projectId?: string;
  type: AccessControlSystemType;
  serialNumber: string;
  locationId?: string;
  location?: ILocation;
  deviceName: string;
  hasReverseInstallation: boolean;
  reader1: IReader;
  reader2?: IReader;
}

export interface IProjectAccessControlSystemByLocation {
  location?: ILocation;
  accessControlSystems: IAccessControlSystem[];
}

export const accessControlSystemTypeMap = {
  [AccessControlSystemType.ENCLOSED_TS]: 'Enclosed TS',
  [AccessControlSystemType.HANDHELD]: 'Handheld',
  [AccessControlSystemType.PORTAL]: 'Open Portal',
  [AccessControlSystemType.TURNSTILE]: 'Turnstile',
};

export const typeOptionList = [
  { value: AccessControlSystemType.PORTAL, label: accessControlSystemTypeMap[AccessControlSystemType.PORTAL] },
  { value: AccessControlSystemType.TURNSTILE, label: accessControlSystemTypeMap[AccessControlSystemType.TURNSTILE] },
  { value: AccessControlSystemType.ENCLOSED_TS, label: accessControlSystemTypeMap[AccessControlSystemType.ENCLOSED_TS] },
  { value: AccessControlSystemType.HANDHELD, label: accessControlSystemTypeMap[AccessControlSystemType.HANDHELD] },
];

export const lifeCycleMap = {
  [LifeCycleType.NEW]: 'New',
  [LifeCycleType.UPGRADE]: 'Upgrade',
  [LifeCycleType.IN_FIELD_TRANSFER]: 'In Field Transfer',
  [LifeCycleType.REFURBISHED]: 'Refurbished',
};

export const lifeCycleOptionList = [
  { value: LifeCycleType.NEW, label: lifeCycleMap[LifeCycleType.NEW] },
  { value: LifeCycleType.UPGRADE, label: lifeCycleMap[LifeCycleType.UPGRADE] },
  { value: LifeCycleType.IN_FIELD_TRANSFER, label: lifeCycleMap[LifeCycleType.IN_FIELD_TRANSFER] },
  { value: LifeCycleType.REFURBISHED, label: lifeCycleMap[LifeCycleType.REFURBISHED] },
];

export const deviceVersionMap = {
  [AccessControlSystemVersion.V1]: 'V1',
  [AccessControlSystemVersion.V2]: 'V2',
  [AccessControlSystemVersion.V3]: 'V3',
  [AccessControlSystemVersion.V3_FACIAL]: 'V3Facial',
};

export const deviceVersionOptionList = [
  { value: AccessControlSystemVersion.V1, label: deviceVersionMap[AccessControlSystemVersion.V1] },
  { value: AccessControlSystemVersion.V2, label: deviceVersionMap[AccessControlSystemVersion.V2] },
  { value: AccessControlSystemVersion.V3, label: deviceVersionMap[AccessControlSystemVersion.V3] },
  { value: AccessControlSystemVersion.V3_FACIAL, label: deviceVersionMap[AccessControlSystemVersion.V3_FACIAL] },
];

export const readerTypeMap = {
  [ReaderType.Entrance]: 'Entrance',
  [ReaderType.Exit]: 'Exit',
  [ReaderType.BiDirectionalByAntenna]: 'BiDirectional By Antenna',
  [ReaderType.BiDirectionalBySpec]: 'BiDirectional By Spec',
};

export const readerTypeByTypeMap = {
  [AccessControlSystemType.PORTAL]: [
    { label: readerTypeMap[ReaderType.Entrance], value: ReaderType.Entrance },
    { label: readerTypeMap[ReaderType.Exit], value: ReaderType.Exit },
    { label: readerTypeMap[ReaderType.BiDirectionalByAntenna], value: ReaderType.BiDirectionalByAntenna },
  ],
  [AccessControlSystemType.TURNSTILE]: [{ label: readerTypeMap[ReaderType.BiDirectionalBySpec], value: ReaderType.BiDirectionalBySpec }],
  [AccessControlSystemType.ENCLOSED_TS]: [{ label: readerTypeMap[ReaderType.BiDirectionalBySpec], value: ReaderType.BiDirectionalBySpec }],
  [AccessControlSystemType.HANDHELD]: [],
};

export const getFallbackAccessControlSystem = (): IAccessControlSystem => ({
  id: null,
  type: AccessControlSystemType.PORTAL,
  serialNumber: null,
  lifeCycle: LifeCycleType.NEW,
  lastRefurbishedDate: null,
  lastMaintenanceDate: null,
  inServiceDate: null,
  warrantyExpirationDate: null,
  orderDate: null,
  vendor: null,
  invoice: null,
  project: null,
  status: null,
  notes: null,
  version: AccessControlSystemVersion.V1,
  reader1: getFallbackReader(),
  reader2: null,
  hasReverseInstallation: false,
});

export const getFallbackReader = (): IReader => ({
  serialNumber: null,
  lastMaintenanceDate: null,
  inServiceDate: null,
  model: null,
  notes: null,
});

export const getFallbackProjectAccessControlSystem = (): IProjectAccessControlSystem => ({
  type: AccessControlSystemType.PORTAL,
  accessControlSystemId: null,
  serialNumber: null,
  deviceName: null,
  hasReverseInstallation: false,
  reader1: null,
  reader2: null,
});
