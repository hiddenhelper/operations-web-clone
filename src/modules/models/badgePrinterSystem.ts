import { IDevice } from './device';
import { INamedEntity } from './general';

export enum PrinterType {
  ZEBRA = 0,
  DATA_CARD = 1,
  BROTHER = 2,
}

export interface IBadgePrintingDevice extends IDevice {
  price: number;
  vendor: string;
  orderDate: string;
  invoice: string;
  model: string;
  inServiceDate: string;
  warrantyExpirationDate: string;
}

export interface ILaptop extends IBadgePrintingDevice {
  osVersion: string;
}

export interface IPrinter extends IBadgePrintingDevice {
  type: PrinterType;
  lastMaintenanceDate: string;
}

export interface IProjectBadgePrintingSystem extends IDevice {
  badgePrintingSystem?: INamedEntity;
  laptopSerialNumber?: string;
  printerSerialNumber?: string;
  scannerSerialNumber?: string;
  project?: {
    id: string;
    name: string;
  };
  shippingDate?: string;
}

export interface IBadgePrintingSystem extends IProjectBadgePrintingSystem {
  name: string;
  laptop: ILaptop;
  printer: IPrinter;
  scanner: IBadgePrintingDevice;
  date?: string;
  index?: number;
}

export interface IBadgePrintingSystemUpdateDate {
  badgePrintingSystemId: string;
  shippingDate: string;
}

export const printerMap = {
  [PrinterType.ZEBRA]: 'Zebra',
  [PrinterType.DATA_CARD]: 'Data Card',
  [PrinterType.BROTHER]: 'Brother',
};

export const getFallbackPrinter = (): IPrinter => ({
  id: null,
  price: null,
  vendor: null,
  orderDate: null,
  invoice: null,
  model: null,
  inServiceDate: null,
  warrantyExpirationDate: null,
  serialNumber: null,
  notes: null,
  lastMaintenanceDate: null,
  type: null,
});

export const getFallbackLaptop = (): ILaptop => ({
  id: null,
  price: null,
  vendor: null,
  orderDate: null,
  invoice: null,
  model: null,
  inServiceDate: null,
  warrantyExpirationDate: null,
  serialNumber: null,
  notes: null,
  osVersion: null,
});

export const getFallbackScanner = (): IBadgePrintingDevice => ({
  id: null,
  price: null,
  vendor: null,
  orderDate: null,
  invoice: null,
  model: null,
  inServiceDate: null,
  warrantyExpirationDate: null,
  serialNumber: null,
  notes: null,
});

export const getFallbackBadgePrinterSystem = (): IBadgePrintingSystem => ({
  id: null,
  notes: null,
  name: null,
  shippingDate: null,
  project: null,
  laptop: getFallbackLaptop(),
  printer: getFallbackPrinter(),
  scanner: getFallbackScanner(),
});
