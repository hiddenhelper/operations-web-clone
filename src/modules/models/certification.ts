import { INamedEntity, GroupValidationType } from './general';

export interface ICertification {
  id: string;
  name: string;
}

export interface IProjectCertification extends ICertification {
  alias: string;
}

export interface ICertificationGroup {
  id: string;
  name: string;
  validationType: GroupValidationType;
  certifications: IProjectCertification[];
}

export interface IWorkerCertificationFile {
  id?: string;
  displayName: string;
  url: string;
  fileSize: number;
}

export interface IWorkerCertification {
  id?: string;
  certificationId?: string;
  idNumber: string;
  projectId?: string;
  certification?: INamedEntity;
  description: string;
  completionDate: string;
  expirationDate: string;
  project?: INamedEntity;
  files?: IWorkerCertificationFile[];
  index?: number;
}

export const getCertificationFallback = (): IWorkerCertification => ({
  id: null,
  certificationId: '',
  idNumber: null,
  projectId: '',
  description: null,
  completionDate: null,
  expirationDate: null,
});
