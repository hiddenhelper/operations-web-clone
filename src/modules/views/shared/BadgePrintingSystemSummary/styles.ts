import { createStyles, makeStyles } from '@material-ui/core/styles';
import { STYLE } from '../../../../constants';
import { toREM } from '../../../../utils/generalUtils';

export const useStyles = makeStyles(theme =>
  createStyles({
    summaryDeviceDetail: {
      color: STYLE.COLOR.SECONDARY_DARKER,
      fontSize: `${toREM(STYLE.FONT.SIZE.PARAGRAPH)}`,
    },
    dropdownIcon: {
      marginLeft: '10px',
      '& .MuiButton-root': {
        minWidth: 0,
      },
      '& .MuiButtonBase-root': {
        padding: 0,

        '& .MuiButton-label': {
          '& .MuiButton-startIcon': {
            margin: 0,
          },
        },
      },
    },
    filterStatusContainer: {
      width: 'auto',
      height: `${toREM(50)}`,
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'flex-end',
      '& .MuiTypography-root': {
        fontSize: STYLE.FONT.SIZE.PARAGRAPH,
      },
    },
    filterStatusSpaced: {
      justifyContent: 'space-between',
    },
    company: {
      height: 'auto',
      display: 'flex',
      flexDirection: 'column',
    },
    deviceAccent: {
      fontWeight: 600,
      paddingBottom: '2px',
    },
    billingModelSecondItem: {
      paddingLeft: 33,
      paddingTop: 37,
    },
    drawerSection: {
      display: 'flex',
      paddingBottom: '3px',
      paddingTop: `${toREM(29)}`,
      wordBreak: 'break-word',
      '& svg': {
        width: 27,
        height: 26,
        '& path': {
          fill: STYLE.COLOR.SECONDARY_DARKER,
        },
      },
    },
    drawerNotesIcon: {
      width: '26.5px !important',
    },
    drawerSecondaryNotesIcon: {
      width: '30.5px !important',
    },
    drawerMonetizationIcon: {
      width: '29px !important',
      height: '26px !important',
    },
    drawerText: {
      display: 'flex',
      flexDirection: 'column',
      wordBreak: 'break-word',
      width: '100%',
      '& span': {
        paddingBottom: '4px',

        '&:last-child': {
          paddingBottom: 0,
        },
      },
    },
    drawerMainText: {
      paddingLeft: `${toREM(39)}`,
      display: 'flex',
      flexDirection: 'column',
      '& span': {
        paddingBottom: '4px',

        '&:last-child': {
          paddingBottom: 0,
        },
      },
    },
    drawerTextSeparation: {
      paddingLeft: `${toREM(15)}`,
    },
    entityTitle: {
      fontSize: `${toREM(STYLE.FONT.SIZE.PARAGRAPH)}`,
    },
    drawerdevices: {
      flexDirection: 'column',
    },
    deviceWrapper: {
      display: 'flex',
      flexDirection: 'row',
      marginBottom: `${toREM(28)}`,

      '&:last-child': {
        marginBottom: 0,
      },
    },
    drawerRowDirection: {
      display: 'flex',
      flexDirection: 'row',
    },
    drawerColumnDirection: {
      display: 'flex',
      flexDirection: 'column',
    },
    drawerOverflowWrapper: {
      overflowY: 'auto',
      position: 'relative',
      paddingBottom: '30px',
      maxHeight: `${toREM(615)}`,
      width: 290,
      paddingRight: '2px',
    },
    bpsIconSize: {
      width: 29,
    },
  })
);
