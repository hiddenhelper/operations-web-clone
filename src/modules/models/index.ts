import * as GeneralModel from './general';
import * as UserModel from './user';
import * as ClientModel from './client';
import * as ProjectModel from './project';
import * as ResourceModel from './resource';
import * as WorkerModel from './worker';
import * as AccessControlSystemModel from './accessControlSystem';
import * as BadgePrintingSystemModel from './badgePrinterSystem';
import * as DeviceModel from './device';
import * as InventoryModel from './inventory';
import * as BadgeModel from './badge';
import * as CertificationModel from './certification';
import * as FileModel from './file';
import * as AddressModel from './address';
import * as ConsentFormModel from './consentForm';
import * as StatisticsModel from './statistics';
import * as TrainingModel from './training';
import * as PaymentModel from './payment';
import * as InvoiceModel from './invoice';
import * as ProcoreModel from './procore';
import * as SearchModel from './search';
import * as ProjectNewModel from './project-new';

export {
  GeneralModel,
  UserModel,
  ClientModel,
  ProjectModel,
  ResourceModel,
  AccessControlSystemModel,
  WorkerModel,
  BadgePrintingSystemModel,
  DeviceModel,
  InventoryModel,
  BadgeModel,
  CertificationModel,
  FileModel,
  AddressModel,
  ConsentFormModel,
  StatisticsModel,
  TrainingModel,
  PaymentModel,
  InvoiceModel,
  ProcoreModel,
  SearchModel,
  ProjectNewModel,
};
