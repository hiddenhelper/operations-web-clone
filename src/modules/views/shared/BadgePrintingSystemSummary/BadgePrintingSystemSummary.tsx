import React, { memo } from 'react';

import Typography from '@material-ui/core/Typography';

import { DeviceStatusIcon, NotesIcon, MonetizationIcon } from '../../../../constants';
import { BadgePrintingSystemModel, DeviceModel, GeneralModel } from '../../../models';
import { getConditionalDefaultValue, getFormattedDate, getDefaultValue } from '../../../../utils/generalUtils';
import { useStyles } from './styles';

export interface IDeviceDrawerProps {
  device: BadgePrintingSystemModel.IBadgePrintingSystem;
  deviceListElement?: React.ReactNode;
}

const BadgePrintingSystemSummary = ({ deviceListElement, device }: IDeviceDrawerProps) => {
  const deviceListHeight = (deviceListElement as any)?.offsetHeight;
  const classes = useStyles({ deviceListHeight: deviceListHeight ?? '100%' });

  return (
    <div className={classes.summaryDeviceDetail}>
      <div className={classes.company}>
        <div className={classes.drawerOverflowWrapper}>
          <div className={classes.drawerText}>
            <Typography className={`${classes.deviceAccent} ${classes.entityTitle}`}>{device.name}</Typography>
            <div className={classes.drawerMainText}>
              <span>{getDefaultValue(device.project?.name, 'No project assigned')}</span>
              <span>{DeviceModel.deviceStatusMap[device.status]}</span>
            </div>
          </div>
          <div className={classes.drawerSection}>
            <NotesIcon className={classes.drawerNotesIcon} />
            <div className={`${classes.drawerText} ${classes.drawerTextSeparation}`}>
              <span>{getDefaultValue(device.notes)}</span>
            </div>
          </div>
          <div className={classes.drawerSection}>
            <div className={classes.drawerText}>
              <Typography className={`${classes.deviceAccent} ${classes.entityTitle}`}>Laptop</Typography>
              <div className={classes.drawerMainText}>
                <span>{getDefaultValue(device?.laptop?.serialNumber)}</span>
                <span>{getDefaultValue(device?.laptop?.model)}</span>
                <span>{getDefaultValue(device?.laptop?.osVersion)}</span>
              </div>
            </div>
          </div>
          <div className={classes.drawerSection}>
            <div className={classes.drawerColumnDirection}>
              <div className={classes.drawerRowDirection}>
                <MonetizationIcon className={`${classes.drawerMonetizationIcon} ${classes.bpsIconSize}`} />
                <div className={`${classes.drawerText} ${classes.drawerTextSeparation}`}>
                  <span>${getDefaultValue(device?.laptop?.price)}</span>
                  <span>{getDefaultValue(device?.laptop?.vendor)}</span>
                  <span>Invoiced {getDefaultValue(device?.laptop?.invoice)}</span>
                </div>
              </div>
            </div>
          </div>
          <div className={classes.drawerSection}>
            <div className={classes.drawerRowDirection}>
              <DeviceStatusIcon className={classes.bpsIconSize} />
              <div className={`${classes.drawerText} ${classes.drawerTextSeparation}`}>
                <span>
                  Order/Purchase Date:{' '}
                  {getConditionalDefaultValue(device?.laptop?.orderDate, getFormattedDate(device?.laptop?.orderDate, GeneralModel.DateFormat.DATE), '')}
                </span>
                <span>
                  Warranty Expiration Date:{' '}
                  {getConditionalDefaultValue(
                    device?.laptop?.warrantyExpirationDate,
                    getFormattedDate(device?.laptop?.warrantyExpirationDate, GeneralModel.DateFormat.DATE),
                    ''
                  )}
                </span>
                <span>
                  In Service Date:{' '}
                  {getConditionalDefaultValue(device?.laptop?.inServiceDate, getFormattedDate(device?.laptop?.inServiceDate, GeneralModel.DateFormat.DATE), '')}
                </span>
              </div>
            </div>
          </div>
          <div className={classes.drawerSection}>
            {device?.laptop?.notes ? (
              <div className={classes.drawerRowDirection}>
                <NotesIcon className={classes.drawerSecondaryNotesIcon} />
                <div className={`${classes.drawerText} ${classes.drawerTextSeparation}`}>
                  <span>{device?.laptop?.notes}</span>
                </div>
              </div>
            ) : (
              <React.Fragment>
                <NotesIcon className={classes.drawerSecondaryNotesIcon} />
                <div className={`${classes.drawerText} ${classes.drawerTextSeparation}`}>
                  <span>-</span>
                </div>
              </React.Fragment>
            )}
          </div>
          <div className={classes.drawerSection}>
            <div className={classes.drawerText}>
              <Typography className={`${classes.deviceAccent} ${classes.entityTitle}`}>Printer</Typography>
              <div className={classes.drawerMainText}>
                <span>{getDefaultValue(BadgePrintingSystemModel.printerMap[device?.printer?.type])}</span>
                <span>{getDefaultValue(device?.printer?.serialNumber)}</span>
                <span>{getDefaultValue(device?.printer?.model)}</span>
              </div>
            </div>
          </div>
          <div className={classes.drawerSection}>
            <div className={classes.drawerColumnDirection}>
              <div className={classes.drawerRowDirection}>
                <MonetizationIcon className={classes.drawerMonetizationIcon} />
                <div className={`${classes.drawerText} ${classes.drawerTextSeparation}`}>
                  <span>${getDefaultValue(device?.printer?.price)}</span>
                  <span>{getDefaultValue(device?.printer?.vendor)}</span>
                  <span>Invoiced {getDefaultValue(device?.printer?.invoice)}</span>
                </div>
              </div>
            </div>
          </div>
          <div className={classes.drawerSection}>
            <div className={classes.drawerRowDirection}>
              <DeviceStatusIcon className={classes.bpsIconSize} />
              <div className={`${classes.drawerText} ${classes.drawerTextSeparation}`}>
                <span>
                  Order/Purchase Date:{' '}
                  {getConditionalDefaultValue(device?.printer?.orderDate, getFormattedDate(device?.printer?.orderDate, GeneralModel.DateFormat.DATE), '')}
                </span>
                <span>
                  Warranty Expiration Date:{' '}
                  {getConditionalDefaultValue(
                    device?.printer?.warrantyExpirationDate,
                    getFormattedDate(device?.printer?.warrantyExpirationDate, GeneralModel.DateFormat.DATE),
                    ''
                  )}
                </span>
                <span>
                  Last Maintenance Date:{' '}
                  {getConditionalDefaultValue(
                    device?.printer?.lastMaintenanceDate,
                    getFormattedDate(device?.printer?.lastMaintenanceDate, GeneralModel.DateFormat.DATE),
                    ''
                  )}
                </span>
                <span>
                  In Service Date:{' '}
                  {getConditionalDefaultValue(
                    device?.printer?.inServiceDate,
                    getFormattedDate(device?.printer?.inServiceDate, GeneralModel.DateFormat.DATE),
                    ''
                  )}
                </span>
              </div>
            </div>
          </div>
          <div className={classes.drawerSection}>
            {device?.printer?.notes ? (
              <div className={classes.drawerRowDirection}>
                <NotesIcon className={classes.drawerSecondaryNotesIcon} />
                <div className={`${classes.drawerText} ${classes.drawerTextSeparation}`}>
                  <span>{device?.printer?.notes}</span>
                </div>
              </div>
            ) : (
              <React.Fragment>
                <NotesIcon className={classes.drawerSecondaryNotesIcon} />
                <div className={`${classes.drawerText} ${classes.drawerTextSeparation}`}>
                  <span>-</span>
                </div>
              </React.Fragment>
            )}
          </div>
          <div className={classes.drawerSection}>
            <div className={classes.drawerText}>
              <Typography className={`${classes.deviceAccent} ${classes.entityTitle}`}>Scanner</Typography>
              <div className={classes.drawerMainText}>
                <span>{getDefaultValue(device?.scanner?.serialNumber)}</span>
                <span>{getDefaultValue(device?.scanner?.model)}</span>
              </div>
            </div>
          </div>
          <div className={classes.drawerSection}>
            <div className={classes.drawerColumnDirection}>
              <div className={classes.drawerRowDirection}>
                <MonetizationIcon className={classes.drawerMonetizationIcon} />
                <div className={`${classes.drawerText} ${classes.drawerTextSeparation}`}>
                  <span>${getDefaultValue(device?.scanner?.price)}</span>
                  <span>{getDefaultValue(device?.scanner?.vendor)}</span>
                  <span>Invoiced {getDefaultValue(device?.scanner?.invoice)}</span>
                </div>
              </div>
            </div>
          </div>
          <div className={classes.drawerSection}>
            <div className={classes.drawerRowDirection}>
              <DeviceStatusIcon className={classes.bpsIconSize} />
              <div className={`${classes.drawerText} ${classes.drawerTextSeparation}`}>
                <span>
                  Order/Purchase Date:{' '}
                  {getConditionalDefaultValue(device?.scanner?.orderDate, getFormattedDate(device?.scanner?.orderDate, GeneralModel.DateFormat.DATE), '')}
                </span>
                <span>
                  Warranty Expiration Date:{' '}
                  {getConditionalDefaultValue(
                    device?.scanner?.warrantyExpirationDate,
                    getFormattedDate(device?.scanner?.warrantyExpirationDate, GeneralModel.DateFormat.DATE),
                    ''
                  )}
                </span>
                <span>
                  In Service Date:{' '}
                  {getConditionalDefaultValue(
                    device?.scanner?.inServiceDate,
                    getFormattedDate(device?.scanner?.inServiceDate, GeneralModel.DateFormat.DATE),
                    ''
                  )}
                </span>
              </div>
            </div>
          </div>
          <div className={classes.drawerSection}>
            {device?.scanner?.notes ? (
              <div className={classes.drawerRowDirection}>
                <NotesIcon className={classes.drawerSecondaryNotesIcon} />
                <div className={`${classes.drawerText} ${classes.drawerTextSeparation}`}>
                  <span>{device?.scanner?.notes}</span>
                </div>
              </div>
            ) : (
              <React.Fragment>
                <NotesIcon className={classes.drawerSecondaryNotesIcon} />
                <div className={`${classes.drawerText} ${classes.drawerTextSeparation}`}>
                  <span>-</span>
                </div>
              </React.Fragment>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};

export default memo(BadgePrintingSystemSummary);
