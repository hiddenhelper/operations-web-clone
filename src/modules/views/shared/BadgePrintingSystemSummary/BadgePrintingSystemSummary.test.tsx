import React from 'react';
import { render } from '@testing-library/react';

import BadgePrintingSystemSummary, { IDeviceDrawerProps } from './BadgePrintingSystemSummary';
import { getFallbackBadgePrinterSystem } from '../../../models/badgePrinterSystem';

describe('BadgePrintingSystemSummary', () => {
  let props: IDeviceDrawerProps;

  beforeEach(() => {
    props = {
      device: getFallbackBadgePrinterSystem(),
    };
  });

  it('should render', () => {
    const wrapper = render(<BadgePrintingSystemSummary {...props} />);
    expect(wrapper.container).toMatchSnapshot();
  });

  it('should print notes', async () => {
    props.device.scanner.notes = 'scanner notes';
    const wrapper = render(<BadgePrintingSystemSummary {...props} />);
    const node = await wrapper.findByText('scanner notes');
    expect(node.textContent).toEqual('scanner notes');
  });
});
