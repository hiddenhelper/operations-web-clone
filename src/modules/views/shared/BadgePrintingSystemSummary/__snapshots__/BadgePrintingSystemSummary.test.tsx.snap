// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgePrintingSystemSummary should render 1`] = `
<div>
  <div
    class="makeStyles-summaryDeviceDetail-1"
  >
    <div
      class="makeStyles-company-5"
    >
      <div
        class="makeStyles-drawerOverflowWrapper-20"
      >
        <div
          class="makeStyles-drawerText-12"
        >
          <p
            class="MuiTypography-root makeStyles-deviceAccent-6 makeStyles-entityTitle-15 MuiTypography-body1"
          />
          <div
            class="makeStyles-drawerMainText-13"
          >
            <span>
              No project assigned
            </span>
            <span />
          </div>
        </div>
        <div
          class="makeStyles-drawerSection-8"
        >
          <svg
            class="makeStyles-drawerNotesIcon-9"
          >
            notes.svg
          </svg>
          <div
            class="makeStyles-drawerText-12 makeStyles-drawerTextSeparation-14"
          >
            <span>
              -
            </span>
          </div>
        </div>
        <div
          class="makeStyles-drawerSection-8"
        >
          <div
            class="makeStyles-drawerText-12"
          >
            <p
              class="MuiTypography-root makeStyles-deviceAccent-6 makeStyles-entityTitle-15 MuiTypography-body1"
            >
              Laptop
            </p>
            <div
              class="makeStyles-drawerMainText-13"
            >
              <span>
                -
              </span>
              <span>
                -
              </span>
              <span>
                -
              </span>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-drawerSection-8"
        >
          <div
            class="makeStyles-drawerColumnDirection-19"
          >
            <div
              class="makeStyles-drawerRowDirection-18"
            >
              <svg
                class="makeStyles-drawerMonetizationIcon-11 makeStyles-bpsIconSize-21"
              >
                monetization.svg
              </svg>
              <div
                class="makeStyles-drawerText-12 makeStyles-drawerTextSeparation-14"
              >
                <span>
                  $
                  -
                </span>
                <span>
                  -
                </span>
                <span>
                  Invoiced 
                  -
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-drawerSection-8"
        >
          <div
            class="makeStyles-drawerRowDirection-18"
          >
            <svg
              class="makeStyles-bpsIconSize-21"
            >
              device-status.svg
            </svg>
            <div
              class="makeStyles-drawerText-12 makeStyles-drawerTextSeparation-14"
            >
              <span>
                Order/Purchase Date:
                 
                
              </span>
              <span>
                Warranty Expiration Date:
                 
                
              </span>
              <span>
                In Service Date:
                 
                
              </span>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-drawerSection-8"
        >
          <svg
            class="makeStyles-drawerSecondaryNotesIcon-10"
          >
            notes.svg
          </svg>
          <div
            class="makeStyles-drawerText-12 makeStyles-drawerTextSeparation-14"
          >
            <span>
              -
            </span>
          </div>
        </div>
        <div
          class="makeStyles-drawerSection-8"
        >
          <div
            class="makeStyles-drawerText-12"
          >
            <p
              class="MuiTypography-root makeStyles-deviceAccent-6 makeStyles-entityTitle-15 MuiTypography-body1"
            >
              Printer
            </p>
            <div
              class="makeStyles-drawerMainText-13"
            >
              <span>
                -
              </span>
              <span>
                -
              </span>
              <span>
                -
              </span>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-drawerSection-8"
        >
          <div
            class="makeStyles-drawerColumnDirection-19"
          >
            <div
              class="makeStyles-drawerRowDirection-18"
            >
              <svg
                class="makeStyles-drawerMonetizationIcon-11"
              >
                monetization.svg
              </svg>
              <div
                class="makeStyles-drawerText-12 makeStyles-drawerTextSeparation-14"
              >
                <span>
                  $
                  -
                </span>
                <span>
                  -
                </span>
                <span>
                  Invoiced 
                  -
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-drawerSection-8"
        >
          <div
            class="makeStyles-drawerRowDirection-18"
          >
            <svg
              class="makeStyles-bpsIconSize-21"
            >
              device-status.svg
            </svg>
            <div
              class="makeStyles-drawerText-12 makeStyles-drawerTextSeparation-14"
            >
              <span>
                Order/Purchase Date:
                 
                
              </span>
              <span>
                Warranty Expiration Date:
                 
                
              </span>
              <span>
                Last Maintenance Date:
                 
                
              </span>
              <span>
                In Service Date:
                 
                
              </span>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-drawerSection-8"
        >
          <svg
            class="makeStyles-drawerSecondaryNotesIcon-10"
          >
            notes.svg
          </svg>
          <div
            class="makeStyles-drawerText-12 makeStyles-drawerTextSeparation-14"
          >
            <span>
              -
            </span>
          </div>
        </div>
        <div
          class="makeStyles-drawerSection-8"
        >
          <div
            class="makeStyles-drawerText-12"
          >
            <p
              class="MuiTypography-root makeStyles-deviceAccent-6 makeStyles-entityTitle-15 MuiTypography-body1"
            >
              Scanner
            </p>
            <div
              class="makeStyles-drawerMainText-13"
            >
              <span>
                -
              </span>
              <span>
                -
              </span>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-drawerSection-8"
        >
          <div
            class="makeStyles-drawerColumnDirection-19"
          >
            <div
              class="makeStyles-drawerRowDirection-18"
            >
              <svg
                class="makeStyles-drawerMonetizationIcon-11"
              >
                monetization.svg
              </svg>
              <div
                class="makeStyles-drawerText-12 makeStyles-drawerTextSeparation-14"
              >
                <span>
                  $
                  -
                </span>
                <span>
                  -
                </span>
                <span>
                  Invoiced 
                  -
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-drawerSection-8"
        >
          <div
            class="makeStyles-drawerRowDirection-18"
          >
            <svg
              class="makeStyles-bpsIconSize-21"
            >
              device-status.svg
            </svg>
            <div
              class="makeStyles-drawerText-12 makeStyles-drawerTextSeparation-14"
            >
              <span>
                Order/Purchase Date:
                 
                
              </span>
              <span>
                Warranty Expiration Date:
                 
                
              </span>
              <span>
                In Service Date:
                 
                
              </span>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-drawerSection-8"
        >
          <svg
            class="makeStyles-drawerSecondaryNotesIcon-10"
          >
            notes.svg
          </svg>
          <div
            class="makeStyles-drawerText-12 makeStyles-drawerTextSeparation-14"
          >
            <span>
              -
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
